air/src/execution_step/instructions/ap/apply_to_arguments.rs

```diff
@@ -30,28 +33,37 @@ pub(super) fn apply_to_arg(
 
     let result = match argument {
         InitPeerId => apply_const(exec_ctx.run_parameters.init_peer_id.as_str(), exec_ctx, trace_ctx),
-        LastError(error_accessor) => apply_last_error(error_accessor, exec_ctx, trace_ctx)?,
+        LastError(error_accessor) => apply_last_error(error_accessor, exec_ctx, trace_ctx),
         Literal(value) => apply_const(*value, exec_ctx, trace_ctx),
         Timestamp => apply_const(exec_ctx.run_parameters.timestamp, exec_ctx, trace_ctx),
         TTL => apply_const(exec_ctx.run_parameters.ttl, exec_ctx, trace_ctx),
         Number(value) => apply_const(value, exec_ctx, trace_ctx),
         Boolean(value) => apply_const(*value, exec_ctx, trace_ctx),
         EmptyArray => apply_const(serde_json::json!([]), exec_ctx, trace_ctx),
-        Scalar(scalar) => apply_scalar(scalar, exec_ctx, trace_ctx, should_touch_trace)?,
-        ScalarWithLambda(scalar) => apply_scalar_wl(scalar, exec_ctx, trace_ctx)?,
-        CanonStream(canon_stream) => apply_canon_stream(canon_stream, exec_ctx, trace_ctx)?,
-        CanonStreamWithLambda(canon_stream) => apply_canon_stream_wl(canon_stream, exec_ctx, trace_ctx)?,
-    };
+        Scalar(scalar) => apply_scalar(scalar, exec_ctx, trace_ctx, should_touch_trace),
+        ScalarWithLambda(scalar) => apply_scalar_wl(scalar, exec_ctx, trace_ctx),
+        CanonStream(canon_stream) => apply_canon_stream(canon_stream, exec_ctx, trace_ctx),
+        CanonStreamWithLambda(canon_stream) => apply_canon_stream_wl(canon_stream, exec_ctx, trace_ctx),
+    }?;
 
     Ok(result)
 }
 
-fn apply_const(value: impl Into<JValue>, exec_ctx: &ExecutionCtx<'_>, trace_ctx: &TraceHandler) -> ValueAggregate {
+fn apply_const(
+    value: impl Into<JValue>,
+    exec_ctx: &ExecutionCtx<'_>,
+    trace_ctx: &TraceHandler,
+) -> ExecutionResult<ValueAggregate> {
     let value = Rc::new(value.into());
     let tetraplet = SecurityTetraplet::literal_tetraplet(exec_ctx.run_parameters.init_peer_id.as_ref());
     let tetraplet = Rc::new(tetraplet);
+    let position = trace_ctx.trace_pos().map_err(|e| {
+        ExecutionError::Uncatchable(UncatchableError::IntConversionError(
```

Comment:
It can be simplified (and similar cases below), as `?` can convert the `UncatchableError` into `ExecutionError`.  From/Into may help further simplify this code, but it seems helper function/method should help to avoid boilerplate.

Reply:
Could you ellaborate where to put the helper?
IMHO I will either has to create 3 separate functions for: ap, canon and call - or to put a function into a common place where it doesn't belong too b/c I can't put the logic into trace_pos().

Reply:
Discussed in person: a helper method for `UncatchableError` and implicit conversion to `ExecutionError`